ui/packages/catalog-ui-search/src/main/webapp/react-component/container/results-export/results-export.tsx

```diff
@@ -0,0 +1,145 @@
+import { exportResult, exportResultSet } from '../../utils/export'
+import { getResultSetCql } from '../../utils/cql'
+import saveFile from '../../utils/save-file'
+
+const contentDisposition = require('content-disposition')
+
+type Result = {
+  id: number
+  source: string
+}
+
+type ExportFormat = {
+  id: string
+  displayName: string
+}
+
+type Props = {
+  selectedResults: Result[]
+}
+
+type State = {
+  downloadDisabled: boolean
+  selectedFormat: string
+  exportFormats: ExportFormat[]
+}
+
+class ResultsExport extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      selectedFormat: 'Select an export option',
+      exportFormats: [],
+      downloadDisabled: true,
+    }
+  }
+  componentDidMount() {
+    this.fetchExportOptions()
+  }
+  fetchExportOptions() {
+    let transformerType = 'metacard'
+
+    if (this.props.selectedResults.length > 1) {
+      transformerType = 'query'
+    }
+
+    fetch(`./internal/transformers/${transformerType}`)
+      .then(response => response.json())
+      .then(exportFormats => {
+        return exportFormats.sort(
+          (format1: ExportFormat, format2: ExportFormat) => {
+            if (format1.displayName > format2.displayName) {
+              return 1
+            }
+
+            if (format1.displayName < format2.displayName) {
+              return -1
+            }
+
+            return 0
+          }
+        )
+      })
+      .then(exportFormats =>
+        this.setState({
+          exportFormats: exportFormats,
+        })
+      )
+  }
+  getResultSources() {
+    return new Set(this.props.selectedResults.map(result => result.source))
+  }
+  getSelectedExportFormatId() {
+    const selectedFormat = this.state.selectedFormat
+    let id = ''
+    this.state.exportFormats.forEach(function(format) {
+      if (format.displayName === selectedFormat) {
+        id = format.id
+      }
+    })
+
+    return id
+  }
+  async onDownloadClick() {
+    const transformerId = this.getSelectedExportFormatId()
+    let response = null
+
+    if (this.props.selectedResults.length > 1) {
+      const cql = getResultSetCql(
+        this.props.selectedResults.map((result: Result) => result.id)
+      )
+      const sources = Array.from(this.getResultSources())
+
+      response = await exportResultSet(transformerId, cql, sources)
+    } else {
+      const source = this.props.selectedResults[0].source
+      const metacardId = this.props.selectedResults[0].id
+
+      response = await exportResult(source, metacardId, transformerId)
+    }
+
+    if (response.status === 200) {
+      const filename = contentDisposition.parse(
+        response.headers.get('content-disposition')
+      ).parameters.filename
+      const contentType = response.headers.get('content-type')
+      const data = await response.text()
+
+      saveFile(filename, 'data:' + contentType, data)
+    }
+  }
+  handleExportOptionChange(name: string) {
```

Comment:
can this be called with `name == undefined`? Not sure if a menu selection can be unselected or not

Reply:
No once you change from the default you have to select a different valid menu selection and this method will never be called if the default is still selected because the download button is disabled until you change from the default.